### PR TITLE
Upload autoyast profile

### PR DESCRIPTION
**Pull request by drpaneas**
It is a good practice for debugging to upload the autoyast
profile which is going to be used for the installation
of the rest of the CaaSP nodes.

- Verification run: http://skyrim.qam.suse.de/tests/71# (you can see `stack_configure-autoyast` at the `assets`)

### CODE REVIEW

File: tests/caasp/stack_configure.pm [hunk 1]
```diff
@@ -38,6 +38,15 @@ sub velum_config {
     mutex_create 'VELUM_CONFIGURED';
 }
 
+# Upload autoyast profile
+sub upload_autoyast {
+    send_key 'alt-tab';    # switch to xterm
+    assert_screen 'xterm';
+    assert_script_run 'curl --insecure --location --remote-name ' . get_var('DASHBOARD_URL') . '/autoyast';
+    upload_logs('./autoyast', failok => 1);
```
Comment (kravciak) [inline tests/caasp/stack_configure.pm]: please rename to "autoyast.xml" and remove "failok => 1" - if there is no profile it's a bug

**Comment by kravciak** [comment]
It's not easy to put it in post_fail_hook because only stack_* modules that are children of caasp_controller have reliable post_fail_hooks.

register_and_check test that is failing was part of MicroOS testsuite and might need few changes to include it properly into stack tests.

In other words - we need special handling for failures because of mutexes, barriers and logs being collected from remote nodes.
See https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/caasp_controller.pm#L43

**Comment by drpaneas** [comment]
@rwx788 I've heard you are experienced with autoyast. Do you have any idea why [this](https://openqa.suse.de/tests/overview?distri=caasp&version=2.0&build=%3A6273%3Azypper.1512734887&groupid=127) succeeds for [master](https://openqa.suse.de/tests/1306472#step/register_and_check/1) and [node2](https://openqa.suse.de/tests/1306473#step/register_and_check/1) but it fails for [node1](https://openqa.suse.de/tests/1306474#step/register_and_check/1) ?  

PS: all of them are using the same `autoyast` profile. So, I don't understand why they have different behavior.

**Comment by rwx788** [comment]
@drpaneas Could you point me to the profile you use. I believe you don't register system using autoyast profile, but it may happen that your local smt handles it. Drop me a message in IRC so we could take a look.

**Comment by okurz** [comment]
@rwx788 you have the plan to also upload the autoyast file in all cases. Can you please make sure we won't duplicate code but apply a common way same for casp and non-casp?

**Comment by rwx788** [comment]
@okurz yes, I've noted this PR to see if we can reuse same code, but I'm non 100% sure about that

**Pull request merged**